portal-api/src/cron-scheduler-jobs/helpers/utilisation-report-helpers.test.ts

```diff
@@ -0,0 +1,637 @@
+        expectedDueDate: new Date('2023-11-07'),
+      },
+      {
+        today: new Date('2023-11-10'),
+        businessDay: '5',
+        bankHolidays: [
+          new Date('2023-11-03'), // Friday
+          new Date('2023-11-06'), // Monday
+        ],
+        // 2023-11-04 and 2023-11-05 are weekend dates.
+        expectedDueDate: new Date('2023-11-09'),
+      },
+    ])(
+      'returns business day $businessDay as due date $expectedDueDate based on bank holidays $bankHolidays',
+      async ({ today, businessDay, bankHolidays, expectedDueDate }) => {
+        // Arrange
+        jest.useFakeTimers().setSystemTime(today);
+        process.env.UTILISATION_REPORT_OVERDUE_CHASER_DATE_BUSINESS_DAYS_FROM_START_OF_MONTH = businessDay;
+        jest.mocked(externalApi.bankHolidays.getBankHolidayDatesForRegion).mockResolvedValue(bankHolidays);
+
+        // Act
+        const dueDate = await getReportOverdueChaserDate();
+
+        // Assert
+        expect(dueDate).toEqual(expectedDueDate);
+      },
+    );
+  });
+
+  describe('getFormattedReportPeriod', () => {
+    it('formats month long report period for as "Month Year"', () => {
+      // Act
+      const result = getFormattedReportPeriod({ start: { month: 1, year: 2035 }, end: { month: 1, year: 2035 } });
+
+      // Assert
+      expect(result).toEqual('January 2035');
+    });
+
+    it('formats report period covering more than one month within one year as "StartMonth to EndMonth EndYear"', () => {
+      // Act
+      const result = getFormattedReportPeriod({ start: { month: 2, year: 2025 }, end: { month: 4, year: 2025 } });
+
+      // Assert
+      expect(result).toEqual('February to April 2025');
+    });
+
+    it('formats report period covering more than one year as "StartMonth StartYear to EndMonth EndYear"', () => {
+      // Act
+      const result = getFormattedReportPeriod({ start: { month: 11, year: 2021 }, end: { month: 1, year: 2022 } });
+
+      // Assert
+      expect(result).toEqual('November 2021 to January 2022');
+    });
+
+    it('formats report period with same start and end month but different years as "StartMonth StartYear - EndMonth EndYear"', () => {
+      // Act
+      const result = getFormattedReportPeriod({ start: { month: 8, year: 2021 }, end: { month: 8, year: 2022 } });
+
+      // Assert
+      expect(result).toEqual('August 2021 to August 2022');
+    });
+  });
+
+  describe('getIsReportDue', () => {
+    const BANK_ID = '123';
+
+    it('throws an error when there are no reports for given period', async () => {
+      // Arrange
+      const reportPeriod = {
+        start: {
+          month: 1,
+          year: 2098,
+        },
+        end: {
+          month: 1,
+          year: 2098,
+        },
+      };
+      jest.mocked(api.getUtilisationReports).mockResolvedValue([]);
+
+      // Act + Assert
+      await expect(async () => await getIsReportDue(BANK_ID, reportPeriod)).rejects.toThrow(
```

Comment:
After some testing, I think we don't need to mark the function inside `expect` with `async-await` because we have used the `rejects` option. I don't think there is necessarily any value in removing the `async-await` part though - just something I found interesting!

Reply:
Ahh nice thanks, I got very confused trying to get this to work in a way that be assert an error was thrown 😅